common/test/TestModule.cs

```diff
@@ -2,16 +2,15 @@
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
 using System;
-using System.Collections.Generic;
-using System.Text;
 using System.Text.RegularExpressions;
 using System.Threading.Tasks;
 
 namespace Microsoft.Azure.Devices.E2ETests
 {
     public class TestModule
     {
-        private Module _module;
+        private static readonly Regex s_hostNameRegex = new Regex("(?<=HostName=).*?(?=;)", RegexOptions.Compiled);
```

Comment:
If you are interested in knowing more about this regex, it employs a negative look behind to require, but not match on, some text ahead of the match we want. Then it also uses a negative look ahead to know when to stop the match.

This way you get exactly what you want, and don't have to parse various matches and index on the one you want.

Also note, I used the non-greedy matcher because there are multiple semi-colons in the connection string, so you want to match to the smallest unit to just get the host name.